crates/primitives/src/specification.rs

```diff
@@ -27,41 +46,34 @@ pub enum SpecId {
     MERGE = 15,           // Paris/Merge	        15537394 (TTD: 58750000000000000000000)
     SHANGHAI = 16,        // Shanghai	            17034870 (TS: 1681338455)
     CANCUN = 17,          // Cancun	                TBD
-    #[cfg(feature = "optimism")]
-    BEDROCK = 128,
-    #[cfg(feature = "optimism")]
-    REGOLITH = 129,
     LATEST = u8::MAX,
-}
-
-impl SpecId {
-    #[inline]
-    pub fn try_from_u8(spec_id: u8) -> Option<Self> {
-        Self::n(spec_id)
-    }
-
-    #[inline]
-    pub const fn enabled(our: SpecId, other: SpecId) -> bool {
-        #[cfg(feature = "optimism")]
-        {
-            let (our, other) = (our as u8, other as u8);
-            let (merge, bedrock, regolith) =
-                (Self::MERGE as u8, Self::BEDROCK as u8, Self::REGOLITH as u8);
-            // If the Spec is Bedrock or Regolith, and the input is not Bedrock or Regolith,
-            // then no hardforks should be enabled after the merge. This is because Optimism's
-            // Bedrock and Regolith hardforks implement changes on top of the Merge hardfork.
-            let is_self_optimism = our == bedrock || our == regolith;
-            let input_not_optimism = other != bedrock && other != regolith;
-            let after_merge = other > merge;
-
-            if is_self_optimism && input_not_optimism && after_merge {
-                return false;
-            }
-        }
+);
 
-        our as u8 >= other as u8
-    }
-}
+#[cfg(feature = "optimism")]
+define_spec_id!(
```

Comment:
I like the idea of having a separate SpecId structure, but it isn't best to wrap it inside macro.

You could do:
```
        /// Specification IDs and their activation block.
        /// 
        /// See: [Ethereum Execution Specs](https://github.com/ethereum/execution-specs)
        #[cfg(feature = "optimism")]
        #[repr(u8)]
        #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Ord, PartialOrd, enumn::N)]
        #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
        pub enum SpecId {
            ...
        }
```

Reply:
Pulled it back out